cmd/kube-apiserver/app/server.go

```diff
@@ -470,6 +471,7 @@ func BuildStorageFactory(s *options.ServerRunOptions) (*serverstorage.DefaultSto
 		return nil, fmt.Errorf("error in initializing storage factory: %s", err)
 	}
 
+	storageFactory.AddCohabitatingResources(networking.Resource("networkpolicies"), extensions.Resource("networkpolicies"))
```

Comment:
flip the order here... the first one in the list is used as the storage version. you need to store as extensions network policies for at least one release to allow rolling apiserver upgrades. next release you can flip the order.